src/osd/OSD.cc

```diff
@@ -6287,6 +6293,50 @@ int OSD::_do_command(
     probe_smart(devid, ds);
   }
 
+  else if (prefix == "clear_cache") {
+    dout(20) << "clearing all caches" << dendl;
+    // Clear the objectstore's cache - onode for Bluestore, pagecache for Filestore
+    r = store->flush_cache();
```

Comment:
I recommend passing a `ostream` here which `flush_cache` can write errors too (like that it can't open `/proc/sys/vm/drop_caches`). This can then be returned in the `MCommandReply` message and displayed in the output of `ceph tell osd.* clear_cache`.

Reply:
The `derr` calls do print out the messages:
```
> ./bin/ceph tell osd.* drop_cache
Error EACCES: 
osd.0: Error flushing objectstore cache: (13) Permission denied
Error EACCES: 
osd.1: Error flushing objectstore cache: (13) Permission denied
```
Would going the `ostream` way cover any case where the `derr` might be missed?

Reply:
I think you're just seeing the messages printed to the terminal from a vstart cluster? (IOW, all of the daemons' stderr is connected to your terminal so you see all of these derr messages.) I don't think it's supposed to be automatic like that.

Reply:
You're right, and it makes sense since these `derr` messages aren't passed around anywhere. I updated the PR.